mlflow/tracing/processor/otel.py

```diff
@@ -35,38 +32,35 @@ def __init__(self, span_exporter: SpanExporter):
             self.span_exporter = span_exporter
         except AttributeError:
             pass
-        self._trace_manager = InMemoryTraceManager.get_instance()
 
-    def on_start(self, span: OTelSpan, parent_context: Context | None = None):
-        """
-        Handle the start of a span. This method is called when an OpenTelemetry span is started.
+        # Only register traces with trace manager when NOT in dual export mode
+        # In dual export mode, MLflow span processors handle trace registration
+        self._should_register_traces = not MLFLOW_TRACE_ENABLE_OTLP_DUAL_EXPORT.get()
+        if self._should_register_traces:
+            self._trace_manager = InMemoryTraceManager.get_instance()
```

Comment:
@B-Step62 Let me know what you think here. Perhaps there's a cleaner approach? Decoupling span processor and trace registration seems like a larger change, but happy to do that if it's straightforward enough :D

(For context, Databricks customers are asking for dual export and export of telemetry statistics as OTLP metrics while simultaneously exporting spans to MLflow's backend)